plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -1339,34 +1339,27 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     model::FileLoc fileLoc;
 
     if (start_.isInvalid() || end_.isInvalid())
-    {
       fileLoc.file = getFile(start_);
-      const std::string& type = fileLoc.file.load()->type;
-      if (type != model::File::DIRECTORY_TYPE && type != _cppSourceType)
-      {
-        fileLoc.file->type = _cppSourceType;
-        _ctx.srcMgr.updateFile(*fileLoc.file);
-      }
-      return fileLoc;
-    }
-
-    clang::SourceLocation realStart = start_;
-    clang::SourceLocation realEnd = end_;
```

Comment:
While trying to figure out a way to calculate metrics for only "internal types", I had a brief excursion into this function to see how type locations (specifically: the file paths) are stored.
Here in the old code I noticed that lines 1344-1350 and 1371-1376 were duplicates, making the function harder to interpret than it actually is. So I took the redundancy out.
Anything that was previously not executed because of the early return statement at line 1350 is now in the 'else' block of that conditional statement. What used to be done by lines 1344-1350 is now simply performed by 1371-1376 for both branches of execution. This should result in the same behavior as before, but with less clutter.